File: src/sql/postgres/protocol/zHelpers.zig
pub fn zCount(slice: []const u8) usize {
    return if (slice.len > 0) slice.len + 1 else 0;
}

pub fn zFieldCount(prefix: []const u8, slice: []const u8) usize {
    if (slice.len > 0) {
        return zCount(prefix) + zCount(slice);
    }

    return zCount(prefix);
}

File: src/sql/postgres/protocol/zHelpers.rs
pub fn z_count(slice: &[u8]) -> usize {
    if !slice.is_empty() {
        slice.len() + 1
    } else {
        0
    }
}

pub fn z_field_count(prefix: &[u8], slice: &[u8]) -> usize {
    if !slice.is_empty() {
        return z_count(prefix) + z_count(slice);
    }

    z_count(prefix)
}

// ported from: src/sql/postgres/protocol/zHelpers.zig
